core/test-runtime/src/system.rs

```diff
@@ -124,7 +124,7 @@ fn execute_block_with_state_root_handler(
 	if let Some(storage_changes_root) = storage_changes_root(header.parent_hash.into()) {
 		digest.push(generic::DigestItem::ChangesTrieRoot(storage_changes_root.into()));
 	}
-	if let Some(new_authorities) = <NewAuthorities>::take() {
+	if let Some(new_authorities) = <NewAuthorities>::get() {
```

Comment:
Hmm, but then this needs to be reset somewhere else?

Reply:
yes you're right, we want to get both root calculation to not include the set value, there is probably more to it, will check later.

Reply:
simply removing the value before root calculation seems fine.